train.py

```diff
@@ -47,33 +47,80 @@ class Config:
     """Total number of training steps."""
     log_to: pathlib.Path = pathlib.Path("./logs")
     learning_rate: float = 3e-4
-
+    # flexible wandb logging
     wandb_project: str = "beetle-traits"
-    wandb_entity: str = "samuelstevens"
+    wandb_entity: str = os.environ.get("WANDB_ENTITY")
     slurm_acct: str = ""
     slurm_partition: str = ""
     n_hours: float = 2.0
 
 
+# @beartype.beartype
+# def make_dataloader(cfg: btx.data.HawaiiConfig):
+#     source = btx.data.HawaiiDataset(cfg)
+#     shuffle = cfg.split == "train"
+#     n_epochs = None if cfg.split == "train" else 1
+#     sampler = grain.samplers.IndexSampler(
+#         num_records=len(source), shuffle=shuffle, num_epochs=n_epochs, seed=cfg.seed
+#     )
+#     ops = [
+#         btx.data.utils.DecodeRGB(),
+#         btx.data.utils.Resize(size=cfg.size),
+#         btx.data.utils.GaussianHeatmap(
+#             size=cfg.size,
+#             sigma=getattr(cfg, "sigma", 3.0),  # fallback if not present on data cfg
+#         ),
+#         grain.transforms.Batch(batch_size=cfg.batch_size, drop_remainder=False),
+#     ]
+#     loader = grain.DataLoader(
+#         data_source=source,
+#         sampler=sampler,
+#         operations=ops,
+#         worker_count=cfg.n_workers,
+#         worker_buffer_size=2,  # per-worker prefetch of output batches
+#         read_options=grain.ReadOptions(num_threads=2, prefetch_buffer_size=64),
+#     )
+
+#     return loader
+
 @beartype.beartype
-def make_dataloader(cfg: btx.data.HawaiiConfig):
-    source = btx.data.HawaiiDataset(cfg)
-    shuffle = cfg.split == "train"
-    n_epochs = None if cfg.split == "train" else 1
+def make_dataloader(cfg: btx.data.HawaiiConfig | object):
```

Comment:
I don't see any reason to change the signature of this function. It should still just accept a btx.data.HawaiiConfig.